x-pack/plugins/log_explorer/public/components/flyout_detail/sub_components/hover_popover_action.tsx

```diff
@@ -9,29 +9,33 @@ import React, { useRef, useState } from 'react';
 import {
   EuiFlexGroup,
   EuiPopover,
-  IconType,
   EuiButtonIcon,
   EuiPopoverTitle,
   EuiToolTip,
+  PopoverAnchorPosition,
 } from '@elastic/eui';
-
-export interface HoverActionType {
-  id: string;
-  tooltipContent: string;
-  iconType: IconType;
-  onClick: () => void;
-  display: boolean;
-}
+import { ValuesType } from 'utility-types';
+import { HoverActionType, useHoverActions } from '../../../hooks/use_hover_actions';
+import { LogDocument } from '..';
 
 interface HoverPopoverActionProps {
   children: React.ReactChild;
-  actions: HoverActionType[];
-  title: string;
+  field: string;
+  value: ValuesType<LogDocument['flattened']>;
+  title?: string;
+  anchorPosition?: PopoverAnchorPosition;
 }
 
-export const HoverActionPopover = ({ children, actions, title }: HoverPopoverActionProps) => {
+export const HoverActionPopover = ({
+  children,
+  title,
+  field,
+  value,
+  anchorPosition = 'upCenter',
+}: HoverPopoverActionProps) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   const leaveTimer = useRef<NodeJS.Timeout | null>(null);
+  const hoverActions: HoverActionType[] = useHoverActions({ field, value });
```

Comment:
**nit:** is this explicit typing necessary? It should already assign it to this type as `useHoverActions` as the return type specified.